types/chartjs-color-string/index.d.ts

```diff
@@ -0,0 +1,29 @@
+// Type definitions for chartjs-color-string 0.6
+// Project: https://github.com/chartjs/chartjs-color-string#readme
+// Definitions by: Ankan Bhattacharya <https://github.com/Ankan002>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare const colorString: colorString.ColorString;
+
+declare namespace colorString {
```

Comment:
There's no reason to use the namespace pattern here; the upstream package just has an export object, which can be typed on DT plainly as:

```ts
export function getRgba(...): ...;
export function getHsla(...): ...;
// ...
```

Reply:
For reference, you only ever need the namespace pattern if the exported object needs to be something that's "merged", usually a function that they've added properties to.

Reply:
Okk but would it support commonjs imports as it has been exported in the format module.exports

Reply:
Yes. The only reason to use `export =` is when the exported thing _isn't_ just a plain object.

Reply:
Okk then we can simply export the object as default export and the interface as a named export

```typescript
export interface ColorString {
        getRgba(colorString?: string): number[] | undefined;
        getHsla(colorString?: string): number[] | undefined;
        getHwb(colorString?: string): number[] | undefined;
        getRgb(colorString?: string): number[] | undefined;
        getHsl(colorString?: string): number[] | undefined;
        getAlpha(colorString?: string): number | undefined;

        hexString(rgba: number[], a?: number): string;
        rgbString(rgba: number[], alpha?: number): string;
        rgbaString(rgba: number[], alpha?: number): string;
        percentString(rgba: number[], alpha?: number): string;
        percentaString(rgba: number[], alpha?: number): string;
        hslString(hsla: number[], alpha?: number): string;
        hslaString(hsla: number[], alpha?: number): string;
        hwbString(hwb: number[], alpha?: number): string;
        keyword(rgb: number[]): string;
    }
    
export default colorString: ColorString;
```

Reply:
Or should I follow this format

```typescript
export function getRgba(...): ...;
export function getHsla(...): ...;
```

@jakebailey

Reply:
The latter, i.e.:

```ts
export function getRgba(colorString?: string): number[] | undefined;
export function getHsla(colorString?: string): number[] | undefined;
export function getHwb(colorString?: string): number[] | undefined;
export function getRgb(colorString?: string): number[] | undefined;
// ...
```

Reply:
Got it

Reply:
@jakebailey just implemented the required changes